### ISSUE TITLE
v2.28.1 - erigon is not syncing with prysm

### ISSUE THREAD

**Issue by Maboperu** [ledgerwatch/erigon#5769]
System information

Erigon version: v2.28.1 

OS & Version: Windows Server 2022

Expected behaviour

erigon is syncing with prysm

Actual behaviour

erigon is not syncing with prysm v3.1.1

**erigon**

logs:

INFO[10-17|17:23:25.146] [2/16 Headers] Handling new payload      height=15768692 hash=0x914c945e6ab9aaef948cf8a8d7325aaba999cd5f5208274078a881f231305c6f
INFO[10-17|17:23:27.448] Commit cycle                             in=2.2989481s
INFO[10-17|17:23:27.452] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-17|17:23:36.542] [2/16 Headers] Handling new payload      height=15768693 hash=0xff10dbba41bfb5ade1313ba050ed45942726bfff455eae68bd3f976d9cfa5ca1
INFO[10-17|17:23:38.845] Commit cycle                             in=2.3010171s
INFO[10-17|17:23:38.848] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-17|17:23:47.593] [2/16 Headers] Handling new payload      height=15768694 hash=0xbebef38c8fec438370bd8bf3bd759b8a350a9a8f43c892dff7a082cc50de698e
INFO[10-17|17:23:49.870] Commit cycle                             in=2.27603s
INFO[10-17|17:23:49.873] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-17|17:24:00.872] [2/16 Headers] Handling new payload      height=15768695 hash=0x94d5ef41dadba85d9be802f0a31fcf0cba6962698a33a53f5c1c61ddfabd60a5
INFO[10-17|17:24:03.154] Commit cycle                             in=2.2800132s
INFO[10-17|17:24:03.159] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-17|17:24:11.551] [2/16 Headers] Handling new payload      height=15768696 hash=0x6abf6bff89416cefad43096e6d9f652ffba7e2e4c0aff6730a984e3bdc00e380
INFO[10-17|17:24:13.823] Commit cycle                             in=2.2709152s
INFO[10-17|17:24:13.828] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-17|17:24:22.794] [txpool] stat                            block=15765689 pending=10000 baseFee=30000 queued=30000 alloc=2.5GB sys=3.2GB

-> txpool is very behind the headers height

configs:

chain : mainnet
port : 30303

datadir : D:\xx
ethash.dagdir : D:\xx

prune : htc
snapshots : true
torrent.download.rate : 20mb

http : true
ws : true 
private.api.addr : localhost:9090

authrpc.jwtsecret : D:xx
authrpc.port : 8551

**prysm**

logs:

[2022-10-17 17:29:48]  INFO blockchain: Called new payload with optimistic block payloadBlockHash=0x5d185a943bfe slot=4933047
[2022-10-17 17:29:48]  INFO blockchain: Called fork choice updated with optimistic block finalizedPayloadBlockHash=0xb9b2ef299133 headPayloadBlockHash=0x5d185a943bfe headSlot=4933047
[2022-10-17 17:29:48]  INFO blockchain: Synced new block block=0x0512a0fa... epoch=154157 finalizedEpoch=154155 finalizedRoot=0x69afa429... slot=4933047
[2022-10-17 17:29:48]  INFO blockchain: Finished applying state transition attestations=70 payloadHash=0x5d185a943bfe slot=4933047 syncBitsCount=508 txCount=140
[2022-10-17 17:29:49]  INFO p2p: Peer summary activePeers=111 inbound=106 outbound=5
[2022-10-17 17:29:56]  WARN powchain: Execution client is not syncing

-> message erigon is not syncing

configs:

mainnet : true 
enable-only-blinded-beacon-blocks : false
accept-terms-of-use : true

datadir : D:\xx 

p2p-max-peers : 120
p2p-host-ip : xx

execution-endpoint : http://localhost:8551
jwt-secret : D:\xx
suggested-fee-recipient : xx

I need urgently help. Because my Validator is not active. :-(

Thank you so much.

**Comment by mwpastore** [comment]
Same here. Also getting these error from prysm: 

```
[2022-10-17 16:29:41]  WARN rpc/validator: Not connected to ETH1. Cannot determine validator ETH1 deposit block number
[2022-10-17 16:29:42] ERROR powchain: Unable to process past deposit contract logs, perhaps your execution client is not fully synced error=Could not process deposit log: received incorrect merkle index: wanted 409932 but got 453707
```

I suspect I went too aggressive on pruning. I'm running erigon with the following arguments: `--datadir=/path/to/foo --prune=htc --prune.r.before=11052984 --torrent.download.rate=100mb`

Thank you in advance for any suggestions!

**Comment by steelep** [comment]
Also having the same issue, also running lighthouse

**Comment by mookim-eth** [comment]
having same problem, running initial sync without snapshot

```
INFO[10-21|10:08:44.599] [Downloader] Downloaded PoS Headers      now=12468 blk/sec=102.533
INFO[10-21|10:09:14.599] [Downloader] Downloaded PoS Headers      now=9585 blk/sec=96.100
INFO[10-21|10:09:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=667.1MB sys=1.3GB
INFO[10-21|10:09:42.357] [Downloader] posAnchor                   blockHeight=8050
INFO[10-21|10:09:44.599] [Downloader] Downloaded PoS Headers      now=8049 blk/sec=51.200
INFO[10-21|10:10:14.591] [p2p] GoodPeers                          eth66=100
INFO[10-21|10:10:14.599] [Downloader] Downloaded PoS Headers      now=4012 blk/sec=134.567
INFO[10-21|10:10:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=803.7MB sys=1.3GB
INFO[10-21|10:10:22.292] [Downloader] posAnchor                   blockHeight=2093
INFO[10-21|10:10:44.625] [2/16 Headers] Flushed buffer file       name=/eth/temp/erigon-sortable-buf-3231967648 alloc=1.1GB sys=1.3GB
INFO[10-21|10:11:14.600] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=765.8MB sys=1.3GB
INFO[10-21|10:12:14.591] [p2p] GoodPeers                          eth66=100
INFO[10-21|10:12:14.610] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=687.3MB sys=1.3GB
INFO[10-21|10:13:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=905.9MB sys=1.3GB
INFO[10-21|10:14:14.592] [p2p] GoodPeers                          eth66=100
INFO[10-21|10:14:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=735.0MB sys=1.3GB
INFO[10-21|10:15:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=821.7MB sys=1.3GB
INFO[10-21|10:16:14.591] [p2p] GoodPeers                          eth66=100
INFO[10-21|10:16:14.600] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=966.3MB sys=1.3GB
INFO[10-21|10:17:14.600] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=915.7MB sys=1.3GB
INFO[10-21|10:18:10.179] [2/16 Headers] etl: temp files removed   total size=8.5GB
INFO[10-21|10:18:10.179] PoS headers verified and saved           requestId=1 fork head=0x657dad735aba8a704173dedf820054e9e5414e71a06b42fe11f41519a99282d2
INFO[10-21|10:18:14.591] [p2p] GoodPeers                          eth66=100
INFO[10-21|10:18:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=782.6MB sys=1.3GB
INFO[10-21|10:18:23.026] [2/16 Headers] DONE                      in=26h34m10.110053712s
INFO[10-21|10:18:32.643] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:18:32.643] [2/16 Headers] Handling new payload      height=15786530 hash=0xd8799b55ad97650caf09706a474596d0f390a22d6906f690883da4b23719b409
INFO[10-21|10:18:42.768] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:18:47.724] [2/16 Headers] Handling new payload      height=15795837 hash=0xc7f2a98cbc45f5700f7a7feb0a1145b6dfb65dbc14512594b6d7701737b07e1a
INFO[10-21|10:18:47.724] [2/16 Headers] Downloading PoS headers... height=15795836 hash=0x996df1cae63ab30798af1271ac08a49b2d3ba6fec95042bd57853395334cf489 requestId=2
INFO[10-21|10:19:14.600] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=423.2MB sys=1.3GB
INFO[10-21|10:19:44.599] [Downloader] Downloaded PoS Headers      now=15795781 blk/sec=1.000
INFO[10-21|10:20:14.591] [p2p] GoodPeers                          eth66=99
INFO[10-21|10:20:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=425.2MB sys=1.3GB
INFO[10-21|10:20:14.599] [Downloader] Downloaded PoS Headers      now=15795752 blk/sec=0.967
INFO[10-21|10:20:41.068] [Downloader] posAnchor                   blockHeight=15795728
INFO[10-21|10:20:44.599] [Downloader] Downloaded PoS Headers      now=15795340 blk/sec=13.733
INFO[10-21|10:21:10.966] PoS headers verified and saved           requestId=2 fork head=0xf13d4d01247a9983a460c33269f143abec381d1144f1a261ee8e88ca3b627d03
INFO[10-21|10:21:11.120] [2/16 Headers] DONE                      in=2m14.705909839s
INFO[10-21|10:21:14.600] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=502.2MB sys=1.3GB
INFO[10-21|10:21:21.419] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:21:21.419] [2/16 Headers] Handling new payload      height=15795837 hash=0xc7f2a98cbc45f5700f7a7feb0a1145b6dfb65dbc14512594b6d7701737b07e1a
INFO[10-21|10:21:30.793] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:21:36.066] [2/16 Headers] Handling new payload      height=15795851 hash=0xbfa17e0192fd6b0c4ef864cb728841c79765d8aaa3f6f77e1822ed7785480708
INFO[10-21|10:21:36.066] [2/16 Headers] Downloading PoS headers... height=15795850 hash=0xfa496a25d74726657d7986529add3355a4a892f161c475b0c893ec8f4ee31208 requestId=3
INFO[10-21|10:21:45.860] PoS headers verified and saved           requestId=3 fork head=0x4f3058cdc3ab6723aead305019593e374d1e557a5bf879c68c686957ade5c1e3
INFO[10-21|10:21:54.791] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:21:54.791] [2/16 Headers] Handling new payload      height=15795851 hash=0xbfa17e0192fd6b0c4ef864cb728841c79765d8aaa3f6f77e1822ed7785480708
INFO[10-21|10:22:04.278] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:22:11.625] [2/16 Headers] Handling new payload      height=15795854 hash=0x11198e9b106f7438d72aeabad74e92c21ee2ac7595a4905d2bcc4207a3cdf27f
INFO[10-21|10:22:11.625] [2/16 Headers] Downloading PoS headers... height=15795853 hash=0xde9488963d05dc119097a6538ad4acc7dee25ef6f3b8a34bebf7a07e33dfb565 requestId=4
INFO[10-21|10:22:14.591] [p2p] GoodPeers                          eth66=100
INFO[10-21|10:22:14.600] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=521.8MB sys=1.3GB
INFO[10-21|10:22:20.563] PoS headers verified and saved           requestId=4 fork head=0x3d6aef8b8cf385f2e422c3f072e64106d1ba80b5f83cc833f1eccfc6a8d69e21
INFO[10-21|10:22:30.016] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:22:30.016] [2/16 Headers] Handling new payload      height=15795854 hash=0x11198e9b106f7438d72aeabad74e92c21ee2ac7595a4905d2bcc4207a3cdf27f
INFO[10-21|10:22:40.082] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:22:48.603] [2/16 Headers] Handling new payload      height=15795857 hash=0x3b9f8f3d6a82b183f7511cd5a3356e673afa7f6d16622aa060f2dbfece8ed8be
INFO[10-21|10:22:48.603] [2/16 Headers] Downloading PoS headers... height=15795856 hash=0x6f532ed6cac6f6dc20ea8e9ab243c9d03169d6cbdf4755145a576980067bab33 requestId=5
INFO[10-21|10:22:57.854] PoS headers verified and saved           requestId=5 fork head=0x96761e7fc8d2a01f1304f508c0029372393884f2d8e884125103c18b46846e22
INFO[10-21|10:23:06.692] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|10:23:06.692] [2/16 Headers] Handling new payload      height=15795857 hash=0x3b9f8f3d6a82b183f7511cd5a3356e673afa7f6d16622aa060f2dbfece8ed8be
INFO[10-21|10:23:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=493.9MB sys=1.3GB
INFO[10-21|10:23:16.017] [2/16 Headers] Waiting for Consensus Layer...

....


INFO[10-21|11:45:14.599] [txpool] stat                            block=0 pending=172 baseFee=0 queued=1235 alloc=560.7MB sys=1.4GB
INFO[10-21|11:45:16.511] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|11:45:23.858] [2/16 Headers] Handling new payload      height=15796270 hash=0xc3236459b670b194ec27361cb05bfaf4c4441fc008261e3097faabfad4925a18
INFO[10-21|11:45:23.858] [2/16 Headers] Downloading PoS headers... height=15796269 hash=0x6cb104fa2392af3050dfd7c5fabcb93144ac87e8ca4a78a8323400f2bdb14e99 requestId=143
INFO[10-21|11:45:33.166] PoS headers verified and saved           requestId=143 fork head=0x40ed2a70f5a034f8966a824840bcbdaf2aa7cf1073b949a12e00f5e396bb9584
INFO[10-21|11:45:41.897] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|11:45:41.897] [2/16 Headers] Handling new payload      height=15796270 hash=0xc3236459b670b194ec27361cb05bfaf4c4441fc008261e3097faabfad4925a18
INFO[10-21|11:45:51.403] [2/16 Headers] Waiting for Consensus Layer...
```

and then it loop `Waiting for Consensus Layer...` forever...

**Comment by mookim-eth** [comment]
re-starting seems ok now: ( i added a log.Info at the beginning of SpawnStageHeaders function)

<details><summary>log</summary>

```
INFO[10-21|11:58:45.293] Reading JWT secret                       path=/eth/jwt.hex
INFO[10-21|11:58:45.293] HTTP endpoint opened for Engine API      url=localhost:8551 ws=true ws.compression=true
INFO[10-21|11:58:45.293] HTTP endpoint opened                     url=172.18.0.1:18545 ws=false ws.compression=true grpc=false
INFO[10-21|11:58:45.298] Started P2P networking                   version=66 self=enode://c29dfd13cacd544ff5dbf0febd7fc3c811ce429a639a0be6d40b8fce2ae5eeb66ddf7b3771dcf81a2d47a38b9b508b8f6d74265697e2c3b51c60fefec9a35c0b@127.0.0.1:30304 name=erigon/v2.28.1-dev-da354bc0/linux-amd64/go1.19.2
INFO[10-21|11:58:45.303] SpawnStageHeaders
INFO[10-21|11:58:45.303] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|11:58:45.313] [txpool] Started
INFO[10-21|11:58:45.988] [2/16 Headers] Handling new payload      height=15796323 hash=0x2ccba0848abbfabde057f5e3c0511465af42aab5e011d6235fed4acde80603dd
INFO[10-21|11:58:45.988] [2/16 Headers] Downloading PoS headers... height=15796322 hash=0xb8e45e789eeb8aa594b4664f5206cedebd8bb33525463f1b3cfb50846f2b4683 requestId=1
INFO[10-21|11:58:56.062] SpawnStageHeaders
INFO[10-21|11:58:56.063] PoS headers verified and saved           requestId=1 fork head=0xbef5d4c5d0492e99986e006d505ff15e5b956f07c7adf9cd5bfbb6a363ff34d4
INFO[10-21|11:59:04.788] SpawnStageHeaders
INFO[10-21|11:59:04.788] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|11:59:04.788] [2/16 Headers] Handling new payload      height=15796323 hash=0x2ccba0848abbfabde057f5e3c0511465af42aab5e011d6235fed4acde80603dd
INFO[10-21|11:59:13.482] SpawnStageHeaders
INFO[10-21|11:59:13.482] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|11:59:24.118] [2/16 Headers] Handling new payload      height=15796339 hash=0x7b2afa29bd80246ca3514413efde0948180044807b809a1a6d8726ee4ea41f02
INFO[10-21|11:59:24.118] [2/16 Headers] Downloading PoS headers... height=15796338 hash=0x05607b67fab43be94a63fc6bb3c4f1b96e046a2d2aa5062281757b1ef4581012 requestId=2
INFO[10-21|11:59:32.783] SpawnStageHeaders
INFO[10-21|11:59:32.784] PoS headers verified and saved           requestId=2 fork head=0x5f862864ae34b95f0c829afd8901f6e35c1741e0d74fe463cba8cf67722d51de
INFO[10-21|11:59:41.330] SpawnStageHeaders
INFO[10-21|11:59:41.330] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|11:59:41.330] [2/16 Headers] Handling new payload      height=15796339 hash=0x7b2afa29bd80246ca3514413efde0948180044807b809a1a6d8726ee4ea41f02
INFO[10-21|11:59:45.295] [txpool] stat                            block=0 pending=194 baseFee=0 queued=1157 alloc=76.0MB sys=139.0MB
INFO[10-21|11:59:49.823] SpawnStageHeaders
INFO[10-21|11:59:49.824] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|11:59:49.824] [2/16 Headers] Downloading PoS headers... height=unknown hash=0xbd196333071f69742881667ef72e4e29f89e82b6c397d7e651c83b9bd0abe981 requestId=3
INFO[10-21|11:59:58.521] SpawnStageHeaders
INFO[10-21|11:59:58.521] PoS headers verified and saved           requestId=3 fork head=0xa741230d8a15571643c92748e0b093a77f02c3d0d618e8a1cf5e0a96955ad1b1
INFO[10-21|12:00:07.057] SpawnStageHeaders
INFO[10-21|12:00:07.057] [2/16 Headers] Waiting for Consensus Layer...
INFO[10-21|12:00:08.815] [p2p] GoodPeers                          eth66=8
INFO[10-21|12:00:45.293] [txpool] stat                            block=0 pending=194 baseFee=0 queued=1157 alloc=81.1MB sys=157.6MB
INFO[10-21|12:00:48.309] [2/16 Headers] Fork choice: chain extension from=0 to=15796341
INFO[10-21|12:01:08.309] [2/16 Headers] write canonical markers   ancestor=9657518 hash=0xf30bb272cf030389cecf6a70ddbe195b8a12f4cf8afbfff25436a1fe44c527e9
INFO[10-21|12:01:28.310] [2/16 Headers] write canonical markers   ancestor=3393444 hash=0x9cac1a7c1b7db68235c38edfbd5e53f07c70c6cfd629d7001cac93b71eda4ff8
INFO[10-21|12:01:43.211] [2/16 Headers] DONE                      in=1m36.153677506s
INFO[10-21|12:01:45.293] [txpool] stat                            block=0 pending=194 baseFee=0 queued=1157 alloc=104.9MB sys=173.8MB
INFO[10-21|12:02:03.211] [3/16 CumulativeIndex] Wrote Cumulative Index gasUsed=538394368849 now=2825078 blk/sec=141253.900
INFO[10-21|12:02:08.814] [p2p] GoodPeers                          eth66=13
INFO[10-21|12:02:23.212] [3/16 CumulativeIndex] Wrote Cumulative Index gasUsed=10832917023769 now=5635903 blk/sec=140541.250
INFO[10-21|12:02:43.212] [3/16 CumulativeIndex] Wrote Cumulative Index gasUsed=29674322416092 now=8463256 blk/sec=141367.650
INFO[10-21|12:02:45.294] [txpool] stat                            block=0 pending=194 baseFee=0 queued=1157 alloc=149.3MB sys=185.8MB
INFO[10-21|12:03:03.212] [3/16 CumulativeIndex] Wrote Cumulative Index gasUsed=55963949487980 now=11264956 blk/sec=140085.000
INFO[10-21|12:03:23.212] [3/16 CumulativeIndex] Wrote Cumulative Index gasUsed=95186383972528 now=14048002 blk/sec=139152.300
INFO[10-21|12:03:36.931] [3/16 CumulativeIndex] DONE              in=1m53.720172313s
INFO[10-21|12:03:44.355] [4/16 BlockHashes] Flushed buffer file   name=/eth/temp/erigon-sortable-buf-227030756 alloc=457.2MB sys=842.5MB
INFO[10-21|12:03:45.293] [txpool] stat                            block=0 pending=194 baseFee=0 queued=1159 alloc=604.1MB sys=844.9MB
INFO[10-21|12:03:51.381] [4/16 BlockHashes] Flushed buffer file   name=/eth/temp/erigon-sortable-buf-346720118 alloc=641.1MB sys=845.6MB
INFO[10-21|12:03:58.553] [4/16 BlockHashes] Flushed buffer file   name=/eth/temp/erigon-sortable-buf-3175976215 alloc=451.9MB sys=851.1MB
INFO[10-21|12:04:05.943] [4/16 BlockHashes] Flushed buffer file   name=/eth/temp/erigon-sortable-buf-258777307 alloc=654.9MB sys=853.2MB
INFO[10-21|12:04:07.423] [4/16 BlockHashes] Flushed buffer file   name=/eth/temp/erigon-sortable-buf-3391974946 alloc=365.0MB sys=854.6MB
INFO[10-21|12:04:08.815] [p2p] GoodPeers                          eth66=19
```
</details>

Seems a `Fork choice: chain extension from=0 to=15796341` is a must for erigon to continue next stages, which is from beacon client

https://github.com/ledgerwatch/erigon/blob/4795217e4e266a8a1b695bb741c51cbfba6489ff/eth/stagedsync/stage_headers.go#L175

```
interrupt, requestId, requestWithStatus := cfg.hd.BeaconRequestList.WaitForRequest(syncing, test)
request := requestWithStatus.Message
forkChoiceMessage, forkChoiceInsteadOfNewPayload := request.(*engineapi.ForkChoiceMessage)

	if forkChoiceInsteadOfNewPayload {
		payloadStatus, err = startHandlingForkChoice(forkChoiceMessage, requestStatus, requestId, s, u, ctx, tx, cfg, test, headerInserter, preProgress)
	} else {
		payloadMessage := request.(*types.Block)
		payloadStatus, err = handleNewPayload(payloadMessage, requestStatus, requestId, s, ctx, tx, cfg, test, headerInserter)
	}
```

As long as prysm do not send a  fork choice message, erigon will wait indefinitely...

**Comment by Maboperu** [comment]
For this matter there are many Issues but no solutions.
Erigon doesn't works for me and I lost money.
And now I use an another execution client and close this post.

**Comment by gufei** [comment]
> re-starting seems ok now: ( i added a log.Info at the beginning of SpawnStageHeaders function)
> 
> log
> Seems a `Fork choice: chain extension from=0 to=15796341` is a must for erigon to continue next stages, which is from beacon client
> 
> https://github.com/ledgerwatch/erigon/blob/4795217e4e266a8a1b695bb741c51cbfba6489ff/eth/stagedsync/stage_headers.go#L175
> 
> ```go
> interrupt, requestId, requestWithStatus := cfg.hd.BeaconRequestList.WaitForRequest(syncing, test)
> request := requestWithStatus.Message
> forkChoiceMessage, forkChoiceInsteadOfNewPayload := request.(*engineapi.ForkChoiceMessage)
> 
> 	if forkChoiceInsteadOfNewPayload {
> 		payloadStatus, err = startHandlingForkChoice(forkChoiceMessage, requestStatus, requestId, s, u, ctx, tx, cfg, test, headerInserter, preProgress)
> 	} else {
> 		payloadMessage := request.(*types.Block)
> 		payloadStatus, err = handleNewPayload(payloadMessage, requestStatus, requestId, s, ctx, tx, cfg, test, headerInserter)
> 	}
> ```
> 
> As long as prysm do not send a fork choice message, erigon will wait indefinitely...

My erigon is using 2.28.1, using prysm, the same Waiting for Consensus Layer...infinite loop forever!  

I would like to know how this problem was handled later? Has it been solved?

Thanks, looking forward to the reply!

**Comment by mookim-eth** [comment]
@gufei Not knowing whether erigon developer is working on this or not...

I re-started from using snapshot provided by archivenode.io, and face the same problem `[ForkChoiceUpdated] stage loop is busy ` like https://github.com/ledgerwatch/erigon/issues/5821

So I just made a try to comment out these three lines

https://github.com/ledgerwatch/erigon/blob/da354bc01a127d26e7c97737dc34806f7e327560/ethdb/privateapi/ethbackend.go#L485-L488

which successfully made erigon to start next stages (body, execution and so on...), but also made more errors, so I changed it back afterwards. I don't know whether this issue is solved by this code change, or just it needs time...

Currently, my erigon works fine (sometimes will lag minutes). I do not need erigon for real-time data usage (like eth staking or front-running), so some lagging is tolerable to me.

Currently, I'm using erigon commit 9711af6ddf01f1eb96c9dfb7bf33c5d69e6f648a
prysm v3.1.2
and using hetzner ax101
